include/seastar/util/modules.hh

```diff
@@ -0,0 +1,30 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2023 ScyllaDB
+ */
+
+#pragma once
+
+#ifndef SEASTAR_MODULE_EXPORT
+#  define SEASTAR_MODULE_EXPORT
+#  define SEASTAR_MODULE_EXPORT_BEGIN
+#  define SEASTAR_MODULE_EXPORT_END
+#  define SEASTAR_BEGIN_INTERNAL_NAMESPACE namespace internal {
```

Comment:
I'd like not to hide C++ keywords from the source file. So if we need to insert something before `namespace internal`, let's keep it in the original source and add only the additions here.

Reply:
okay. instead of using 
```c++
SEASTAR_BEGIN_INTERNAL_NAMESPACE
// details go here
SEASTAR_MODULE_EXPORT_END
```
in the next revision, will have something like
```c++
SEASTAR_MODULE_EXPORT_END
namespace internal {
// details go here
}
SEASTAR_MODULE_EXPORT_BEGIN
```